force-app/main/default/classes/RecurringDonationsSelector.cls

```diff
@@ -0,0 +1,81 @@
+/*
+    Copyright (c) 2022 Salesforce.org
+    All rights reserved.
+
+    Redistribution and use in source and binary forms, with or without
+    modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+
+/**
+* @author Salesforce.org
+* @date 2022
+* @description Recurring Donations Selector class.
+*/
+public inherited sharing class RecurringDonationsSelector {
+
+    @TestVisible
+    private Id currentContactId {
+        get {
+            if(currentContactId == null) {
+                currentContactId = getCurrentContactId();
+            }
+            return currentContactId;
+        }
+        set;
+    }
+
+    /**
+     * @description Returns the Recurring Donation records related to the Experience Site user.
+     * @return List<npe03__Recurring_Donation__c>
+     */
+    public List<npe03__Recurring_Donation__c> getRecurringDonations() {
```

Comment:
This function can be moved to RD2_QueryService

Reply:
I prefer to have the LWC controller with the aura methods, the Service with the processes/business logic, and the selector with the queries to the DB. Moving this method to the controller implies to have this method with the process to create the list of recurringDonationView objects in the controller layer instead of the Service layer.